### ISSUE TITLE
chessboard effect is very clear when training a 2X model

### ISSUE THREAD

**Issue by deepxiangfa** [alexjc/neural-enhance#68]
The model is trained with MSCOCO data set and the enhanced image has the chessboard effect,  anyone encounter the same problem?

![enhanced](https://cloud.githubusercontent.com/assets/22540620/20162927/74da3d42-a733-11e6-9e12-10939f0da133.jpg)

**Comment by eeric** [comment]
how much is the size of the training sample? Contains ambiguous images? Contains clear images? Is there a label for the sample?

**Comment by deepxiangfa** [comment]
Thanks for replying. I used  the MSCOCO dataset, most of the images in the dataset is around 480by640, and for training, I used 96 by 96 patch(the batch-shape parameter is 96). The above image is the result  after 8 epochs of training. what do you mean by ambiguous images and clear images?

**Comment by eeric** [comment]
I amn't author.
It was generated by downscaled or bicubic interpolation to a 96*96 sample for training data?

**Comment by alexjc** [comment]
@deepxiangfa Yes, it happens if you pick certain parameters. It's an efficient way for the network to generate output that looks acceptable if you don't zoom in.

Try increasing the smoothness level when pre-training (perceptual layer 2_2), then the adversary weight should take care of it in the next stage of training (perceptual layer 5_2).

**Comment by bloc97** [comment]
@alexjc Is this checkerboard pattern related to deconvolution?
Chris Olah did a blog entry on this...
http://distill.pub/2016/deconv-checkerboard/

**Comment by bloc97** [comment]
If I remember the solution was to use resize-convolution instead of simple deconvolution.

**Comment by alexjc** [comment]
@block97 I tried the resize-convolution on this codebase and results were significantly worse than just tuning the smoothness (total variation loss) with the same network capacity.  My current understanding is that the only advantage of resize-convolution is that it doesn't require tuning one parameter.

**Comment by bloc97** [comment]
@alexjc I believe that using the right stride/padding combinations it is possible to reduce the artifacts, what parameters did @deepxiangfa use?

**Comment by alexjc** [comment]
@bloc97 It's doing sub-pixel convolution (different 3x3 input kernels for each pixel of 2x2 output) so there will always be potential mismatches between pixels during upscaling, which is the reason for the total-variation loss to prevent this.

**Comment by deepxiangfa** [comment]
@bloc97  Yes, I trained the network without using TV loss, I will add the TV loss and train the network again to see the result.

**Comment by alexjc** [comment]
The scripts used to train the network in Releases are now in the `train/` folder.
